check_source.py

```diff
@@ -206,17 +206,28 @@ def check_source_submission(self, source_project, source_package, source_revisio
 
         return True
 
-    def suppresses_whitelist_warnings( self, source_project, source_package):
+
+    def suppresses_whitelist_warnings( self, source_project, source_package ):
         # checks if there's a rpmlintrc that suppresses warnings that we check
         found_entries = set()
         contents = source_file_load(self.apiurl, source_project, source_package, source_package+'-rpmlintrc')
         if contents:
+            contents = re.sub(r'(?m)^ *#.*\n?', '', contents)
             matches = re.findall(r'addFilter\(["\']([^"\']+)["\']\)', contents)
+            # this is a bit tricky. Since users can specify arbitrary regular expresions it's not easy
+            # to match bad_rpmlint_entries against what we found
             for entry in self.bad_rpmlint_entries:
                 for match in matches:
-                    if match.startswith(entry):
-                        self.logger.info(f'found suppressed whitelist warning: {entry}')
-                        found_entries.add(entry)
+                    # First we try to see if our entries appear verbatim in the rpmlint entries
+                    if entry in match:
+                        self.logger.info(f'found suppressed whitelist warning: {match}')
+                        found_entries.add(match)
+                    # then we check if one of the entries in the rpmlint file would match one
+                    # of our entries (e.g. addFilter(".*")
```

Comment:
nit: comment does not really make sense with the example as `.*` would match `.*` verbatim.

Reply:
.* would match any entry in the list (with the new logic it doesn't contain entries for .* anymore). Not sure if I understand you correctly. The example is for an rpmlint entry that would cause the branch to be taken

Reply:
Yes, I see it now was thinking the re matched the other direction.